lib/types.nix

```diff
@@ -287,6 +287,12 @@ rec {
       merge = mergeEqualOption;
     };
 
+    nonEmptyStr = elemType:
```

Comment:
You don't need `elemType`. `nonEmptyStr` is already a type and doesn't take a type argument.